File: 14-longest-common-prefix/longest-common-prefix.java
class Solution {
   public String longestCommonPrefix(String[] strs) {
        if (strs.length == 0 || strs[0] == "") return "";
        String longest = "";
        int n = strs[0].length();

        for(int i = 0; i < strs.length; i++) {
            n = Math.min(n, strs[i].length());
        }

        for (int i = 0; i < n; i++) {
            boolean valid = true;
            
            char c = strs[0].charAt(i);

            for (int j = 0; j < strs.length; j++) {
                if (strs[j].charAt(i) != c) {
                    valid = false; break;
                }
            }
            if (valid) {
                longest += c;
            } else {
                break;
            }
        }     
        return longest;
    }
}

File: 14-longest-common-prefix/longest-common-prefix.swift
class Solution {
    func longestCommonPrefix(_ strs: [String]) -> String {
        var longest: String = "", idxs = strs[0].indices
        if strs.count == 0 { return ""}
        if strs.count == 1 { return strs[0] }
        for i in idxs {
            for j in 1..<strs.count {
                var prev: String = strs[j-1], curr: String = strs[j]
                if i < prev.endIndex && i < curr.endIndex && prev[i] == curr[i] {

                } else { return longest }
            }
            longest += String(strs[0][i])
        }
        return longest
    }
}